File: mergeSort/merge_sort.py
from typing import List

class merge_sort:
    def __init__(self, num: int, strings: List[str]):
        self.num = num
        self.strings = strings

        for i in range(self.num):
            print(self.sort(strings=self.strings[i].split(" ")))
    
    def sort(self, strings: List[str]) -> list:
        if len(strings) > 1:
            mid = len(strings)//2

            left = strings[:mid]
            right = strings[mid:]

            self.sort(strings=left)
            self.sort(strings=right)

            i = j = k = 0

            while i < len(left) and j < len(right):
                if len(left[i]) < len(right[j]):
                    strings[k] = left[i]
                    i += 1
                elif len(left[i]) > len(right[j]):
                    strings[k] = right[j]
                    j += 1
                else:
                    strings[k] = left[i]
                    i += 1
                k += 1
            
            while i < len(left) or j < len(right):
                if i < len(left):
                    strings[k] = left[i]
                    i += 1
                elif j < len(right):
                    strings[k] = right[j]
                    j += 1
                k += 1
        return strings

if __name__ == "__main__":
    merge_sort(num=3, strings=[
        "ab cd e j asd ljffg df",
        "a a b b c c",
        "xy yx zxy zx xzy xxx",
    ])

File: mergeSort/mergeSort.js
class MergeSort {
    constructor(num, strings) {
        this.num = num;

        for (let i=0; i<num; i++) {
            this.string = strings[i].split(' ');
            console.log(sort());
        }
    }

    sort() {
        if (this.string.length > 1) {
            let mid = Math.floor(this.string.length/2);

            let left = this.string.slice(0,mid);
            let right = this.string.slice(mid,strings.length);

            left = this.sort(left);
            right = this.sort(right);

            let i = 0;
            let j = 0;
            let k = 0;

            while (i < left.length && j < right.length) {
                if (left.length < right.length) {
                    this.string[k] = left[i];
                    i++;
                }
                else if (left.length > right.length) {
                    this.string[k] = right[j];
                    j++;
                }
                else {
                    this.string[k] = left[i];
                    i++;
                }
                k++;
            }

            while (i < left.length || j < right.length) {
                if (i < left.length) {
                    this.string[k] = left[i];
                    i++;
                }
                else if (j < right.length) {
                    this.string[k] = right[j];
                    j++;
                }
                k++;
            }
        }
        return this.string
    }
}

new MergeSort(3, [
    "ab cd e j asd ljffg df",
    "a a b b c c",
    "xy yx zxy zx xzy xxx",
])